frontend/lib/activation/deeplink_activation.dart

```diff
@@ -0,0 +1,211 @@
+            body: Padding(
+                padding: const EdgeInsets.all(20),
+                child: Column(
+                  children: [
+                    Padding(
+                      padding: const EdgeInsets.all(20),
+                      child: status == DeepLinkActivationStatus.invalidLink
+                          ? null
+                          : Text(t.activation.description,
+                              style: theme.textTheme.headlineSmall, textAlign: TextAlign.center),
+                    ),
+                    Center(
+                        child: cardInfo != null
+                            ? IdCard(
+                                cardInfo: cardInfo,
+                                region: region != null ? Region(region.prefix, region.name) : null,
+                                // We trust the backend to have checked for expiration.
+                                isExpired: false,
+                                isNotYetValid: false,
+                              )
+                            : null),
+                    Padding(
+                      padding: const EdgeInsets.all(32),
+                      child: Column(
+                        mainAxisSize: MainAxisSize.min,
+                        children: [
+                          ...switch (status) {
+                            DeepLinkActivationStatus.invalidLink => [_WarningText(t.activation.invalidCode)],
+                            DeepLinkActivationStatus.limitReached => [
+                                _WarningText('${t.activation.limitReached} ($cardsInUse/$maxCardAmount)')
+                              ],
+                            DeepLinkActivationStatus.alreadyExists => [_WarningText(t.activation.alreadyExists)],
+                            DeepLinkActivationStatus.valid => [
+                                ElevatedButton(
+                                  onPressed: activationCode != null
+                                      ? () {
+                                          activateCard(context, activationCode);
+                                        }
+                                      : null,
+                                  child: Text(t.activation.buttonText),
+                                )
+                              ],
+                          },
+                        ],
+                      ),
+                    ),
+                  ],
+                )),
+          );
+        });
+  }
+}
+
+class _WarningText extends StatelessWidget {
+  final String text;
+
+  const _WarningText(this.text);
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Icon(Icons.warning, color: Theme.of(context).colorScheme.secondary),
+        Text(text, textAlign: TextAlign.center)
+      ],
+    );
+  }
+}
+
+DynamicActivationCode? getActivationCode(BuildContext context, String base64qrcode) {
+  try {
+    if (base64qrcode.length % 4 == 0) {
```

Comment:
there is an issue if you try to convert this code if its not multiple of 4 thats why this check is needed

Reply:
Did you URLencode the base64 string? The `=` chars need to be escaped.

Reply:
I updated the `console.log()` in the druckerei. in a separate task a correct links has to be generated and send via mail/printed on pdf
The router decodes the string properly
This is how it should look like
```js
CmMKOQoYQW5kcmVhcyBGaXNjaGVyIERlZXBsaW5rEImdARoZCgIIXRIDCIg5GggI%2B6CpnwEQASoECJuaARIQIEQAaBcbDba2KBLYakIzTxoU93ZyQ78v3kpAiDUQDMsiMPYDMXg%3D
```

Reply:
I think we should check here not only for the length, but also it is a valid base64 in gerneal, so it also is not allowed to include special chars. But the "convert" method ot the Base64Decoder already does it, so we only need to listen for the errors it may throw.

But at least this boolean expression should be named, so we know what is happening here, when reading the code, e.g. 
`bool isBase64WithValidLength = base64qrcode.length % 4 == 0 `
But i think you should make sure you have the correct

Reply:
I agree, that it should be enough to catch FormatException thrown by [Base64Decoder](https://api.flutter.dev/flutter/dart-convert/Base64Decoder-class.html) instead of checking `length % 4 == 0` and silently failing.

Reply:
added a format exception